### PR TITLE
✨ display total of votes

### PR DESCRIPTION
**Pull request by patou**
Display total of votes

image

Publish after the current campaign.

### CODE REVIEW

File: functions/src/stats-manager.ts
Comment (hgwood) [inline functions/src/stats-manager.ts:47]:
```suggestion
            total: counters.total + 1
```

File: functions/src/update-stats.ts
Comment (hgwood) [inline functions/src/update-stats.ts:32]: Shouldn't there be a +1 here?

File: ui/src/index.ts
Comment (hgwood) [inline ui/src/index.ts:134]: What makes this needed now but not before? Can't TypeScript infer the type of snapshot from the type of stats?

File: ui/src/index.ts
Comment (hgwood) [inline ui/src/index.ts:147]: Same questions as above.

File: ui/src/services/renderTemplate.ts
Comment (hgwood) [inline ui/src/services/renderTemplate.ts:51]: Very nice! I like that it falls back on the sum if the database does not have the result already.

One suggestion though: if total is zero, then the sum is computed. This could be avoided like this:
```suggestion
                    typeof row.counts.total === "number"
                      ? row.counts.total
                      : keys.reduce(
                          (total, key) => total + (row.counts[key] || 0),
                          0
                        )
```

**Review by hgwood — COMMENTED** [review]
Thanks a lot @patou !

File: ui/src/index.ts
Comment (patou) [reply ui/src/index.ts:134]: I had warning about it to write explicitly "any"

File: functions/src/update-stats.ts
Comment (patou) [reply functions/src/update-stats.ts:32]:
```suggestion
      total: previousCounters.total + 1
```

**Review by hgwood — APPROVED** [review]

**Comment by hgwood** [comment]
I'm gonna check that "any" thing before merging.

**Comment by hgwood** [comment]
Looks like your warning you saw was local only. My local env and CI are fine with it.

**Comment by hgwood** [comment]
We should wait for the end of the current survey to end before we merge, otherwise, the total will be off.

**Comment by patou** [comment]
> Publish after the current campaign.

yes I say it in the description : Publish after the current campaign.

**Pull request merged**
